### PR TITLE
{data}[foss/2020b] NCL v6.6.2

### PR DESCRIPTION
**Pull request by kosl**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by kosl** [comment]
Bump

File: easybuild/easyconfigs/g/g2clib/g2clib-1.6.3-foss-2020b.eb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name = 'g2clib'
+version = '1.6.3'
+
+homepage = 'https://www.nco.ncep.noaa.gov/pmb/codes/GRIB2/'
+description = """Library contains GRIB2 encoder/decoder ('C' version)."""
+
+toolchain = {'name': 'foss', 'version': '2020b'}
```
Comment (migueldiascosta) [inline easybuild/easyconfigs/g/g2clib/g2clib-1.6.3-foss-2020b.eb:7]:
```suggestion
toolchain = {'name': 'GCCcore', 'version': '10.2.0'}
```

and change the filename

File: easybuild/easyconfigs/g/g2lib/g2lib-3.2.0-GCCcore-10.2.0.eb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+homepage = 'https://www.nco.ncep.noaa.gov/pmb/codes/GRIB2/'
+description = """Library contains GRIB2 encoder/decoder and search/indexing routines."""
+
+toolchain = {'name': 'GCCcore', 'version': '10.2.0'}
+
+source_urls = [homepage]
+sources = ['%(name)s-%(version)s.tar']
+patches = [
+    '%(name)s-%(version)s_compilation.patch',
+    # '%(name)s-%(version)s_makefile.patch',
```
Comment (migueldiascosta) [inline easybuild/easyconfigs/g/g2lib/g2lib-3.2.0-GCCcore-10.2.0.eb:13]: why is this no longer needed? (compared to `g2lib-3.2.0-GCCcore-10.3.0.eb`)

File: easybuild/easyconfigs/g/g2lib/g2lib-3.2.0_compilation.patch [hunk 3]
```diff
@@ -0,0 +1,1145 @@
+diff -Nru g2lib-3.2.0-orig/addfield.f g2lib-3.2.0/addfield.f
```
Comment (migueldiascosta) [inline easybuild/easyconfigs/g/g2lib/g2lib-3.2.0_compilation.patch:1]: please add description of patch and author here

File: easybuild/easyconfigs/g/g2lib/g2lib-3.2.0_compilation.patch [hunk 4]
```diff
@@ -0,0 +1,1145 @@
...
+       if ( numcoord .ne. 0 ) then
+         allocate (coordieee(numcoord))
+-        call g2_gbytesc(cgrib,coordieee,iofst,32,0,numcoord)
++        call g2_gbytesc(cgrib,int(coordieee),iofst,32,0,numcoord)
+         call rdieee(coordieee,coordlist,numcoord)
+         deallocate (coordieee)
+         iofst=iofst+(32*numcoord)
+diff -Nru g2lib-3.2.0-orig/getfield.f~ g2lib-3.2.0/getfield.f~
+--- g2lib-3.2.0-orig/getfield.f~	1970-01-01 01:00:00.000000000 +0100
++++ g2lib-3.2.0/getfield.f~	2021-03-18 19:20:40.734837860 +0100
```
Comment (migueldiascosta) [inline easybuild/easyconfigs/g/g2lib/g2lib-3.2.0_compilation.patch:51]: this was a mistake, no? it's adding an editor temporary file

File: easybuild/easyconfigs/g/g2lib/g2lib-3.2.0_compilation.patch [hunk 5]
```diff
@@ -0,0 +1,1145 @@
...
+       elseif (idrsnum.eq.50) then ! Spectral simple
+         call simunpack(cgrib(ipos),lensec-5,idrstmpl,ndpts-1,
+-     & fld(2))
++     & fld(2:ndpts))
+         ieee=idrstmpl(5)
+         call rdieee(ieee,fld(1),1)
+       elseif (idrsnum.eq.51) then ! Spectral complex
+diff -Nru g2lib-3.2.0-orig/gf_unpack7.f~ g2lib-3.2.0/gf_unpack7.f~
+--- g2lib-3.2.0-orig/gf_unpack7.f~	1970-01-01 01:00:00.000000000 +0100
++++ g2lib-3.2.0/gf_unpack7.f~	2020-12-16 18:58:40.000000000 +0100
```
Comment (migueldiascosta) [inline easybuild/easyconfigs/g/g2lib/g2lib-3.2.0_compilation.patch:896]: same here

File: easybuild/easyconfigs/g/g2lib/g2lib-3.2.0_compilation.patch [hunk 6]
```diff
@@ -0,0 +1,1145 @@
...
+ 
+       ieee = idrstmpl(1)
+-      call rdieee(ieee,ref,1)
++      call rdieee(char(ieee),ref,1)
+       bscale = 2.0**real(idrstmpl(2))
+       dscale = 10.0**real(-idrstmpl(3))
+       nbits = idrstmpl(4)
+diff -Nru g2lib-3.2.0-orig/specunpack.f~ g2lib-3.2.0/specunpack.f~
+--- g2lib-3.2.0-orig/specunpack.f~	1970-01-01 01:00:00.000000000 +0100
++++ g2lib-3.2.0/specunpack.f~	2020-12-16 18:58:41.000000000 +0100
```
Comment (migueldiascosta) [inline easybuild/easyconfigs/g/g2lib/g2lib-3.2.0_compilation.patch:1037]: same here

File: easybuild/easyconfigs/n/NCL/NCL-6.6.2-foss-2020b.eb [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    ('g2lib', '3.2.0'),
+    ('g2clib', '1.6.3'),
+    ('HDF', '4.2.15'),
+    ('HDF5', '1.10.7'),
+    ('netCDF', '4.7.4'),
+    ('netCDF-Fortran', '4.5.3'),
+    ('Szip', '2.1.1'),
+    ('freetype', '2.10.3'),
+    ('zlib', '1.2.11'),
+    # ('GDAL', '3.9.1'),
```
Comment (migueldiascosta) [inline easybuild/easyconfigs/n/NCL/NCL-6.6.2-foss-2020b.eb:44]: why is `GDAL` commented?

File: easybuild/easyconfigs/n/NCL/NCL-6.6.2-foss-2020b.eb [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    ('UDUNITS', '2.2.26'),
+    ('ESMF', '8.0.1'),
+    ('bzip2', '1.0.8'),
+    ('cairo', '1.16.0'),
+    ('libiconv', '1.16'),
+    ('GSL', '2.6'),
+    ('libpng', '1.6.37'),
+    ('libjpeg-turbo', '2.0.5'),
+    # ('pixman', '0.40.0'),
```
Comment (migueldiascosta) [inline easybuild/easyconfigs/n/NCL/NCL-6.6.2-foss-2020b.eb:53]: `pixman` is a dependency of `cairo`, so wouldn't be needed here in any case - delete commented line?

**Review by migueldiascosta — CHANGES_REQUESTED** [review]

**Comment by lexming** [comment]
@kosl Thanks for your contribution. The easyconfig for NCL/6.6.2-foss-2020b has already been merged in #15820. So I'll close this PR.

**Pull request closed without merging**